packages/flutter_tools/lib/src/macos/cocoapod_utils.dart

```diff
@@ -14,20 +14,51 @@ import '../project.dart';
 Future<void> processPodsIfNeeded(
   XcodeBasedProject xcodeProject,
   String buildDirectory,
-  BuildMode buildMode) async {
+  BuildMode buildMode, {
+  bool forceCocoaPodsOnly = false,
+}) async {
   final FlutterProject project = xcodeProject.parent;
+
+  // When using Swift Package Manager, the Podfile may not exist so if there
+  // isn't a Podfile, skip processing pods.
+  if (project.usesSwiftPackageManager && !xcodeProject.podfile.existsSync() && !forceCocoaPodsOnly) {
+    return;
+  }
   // Ensure that the plugin list is up to date, since hasPlugins relies on it.
-  await refreshPluginsList(project, macOSPlatform: project.macos.existsSync());
-  if (!(hasPlugins(project) || (project.isModule && xcodeProject.podfile.existsSync()))) {
+  await refreshPluginsList(
+    project,
+    macOSPlatform: project.macos.existsSync(),
+    forceCocoaPodsOnly: forceCocoaPodsOnly,
+  );
+
+  // If there are no plugins and if the project is a not module with an existing
+  // podfile, skip processing pods
+  if (!hasPlugins(project) && !(project.isModule && xcodeProject.podfile.existsSync())) {
```

Comment:
I change this if statement to try and make it easier to read, it kept confusing me lol. I also added unit tests for this file/function to make sure it remained the same. I guess it was never unit tested directly before...
https://github.com/flutter/flutter/blob/503ae5b9324028d3c4bad797873a9d13469ab33b/packages/flutter_tools/test/general.shard/macos/cocoapod_utils_test.dart